TACC/Core-CMS
# feat: CMD-97 add content around news list

## Overview

Add content around article list on News page.

> [!TIP]
> Technically, we _could_ do this via [TACC/Core-CMS-Plugin-Remote-Content](https://github.com/TACC/Core-CMS-Plugin-Remote-Content) since #868, **but** page might load slower.

## Related

- [CMD-97](https://tacc-main.atlassian.net/browse/CMD-97)
- alternative to #869

## Changes

Depends on commit.

## Testing

1. Have a CMS with News app installed and working.
2. Open http://localhost:8000/.
3. View page Structure.
4. Verify nothing is changed.
5. Open http://localhost:8000/news/.
6. View page Structure.
7. Verify "Content" placeholder is available.
8. Add "Text" plugin, with some text, to "Content".
9. Verify your text is visible on the page.
10. Verify article list and title is still visible on the page.
11. 🆕 Visit an article page.
12. 🆕 ❌ Verify your text is **not** visible on the page.

## UI

### Solution C

#### **C.3.** Add "Content" to News (fa9b84d)

> [!CAUTION]
> - Same content appears on News List **and** News Article pages.
> - Admin requires Dev to move content (via template or CSS).

| Home | News |
| - | - |
| <img width="1194" alt="C3 Add Content to News - Home (No Change)" src="https://github.com/user-attachments/assets/fce71f4a-d39b-4bc3-a312-43f2905b2c7b"> | <img width="1194" alt="C3 Add Content to News - News (New Placeholder)" src="https://github.com/user-attachments/assets/db07451a-a214-4ce6-85e4-661ea3dca89d"> |

#### **C.2.** Add 4 New Placeholders to All Pages (similar to 8187a20)

> [!CAUTION]
> - Placeholder available on all pages.
> - Same content appears on News List **and** News Article pages.
> - Admin requires dev to move content (via template or CSS).

| Home | News |
| - | - |
| <img width="1194" alt="C2 Positional Placeholders - Home" src="https://github.com/user-attachments/assets/bf2ff7e1-6708-4088-ad07-b27bdc84fc87"> | <img width="1194" alt="C2 Positional Placeholders - news" src="https://github.com/user-attachments/assets/5ed779c4-726d-4759-bf2b-70a0fe87bb52"> |

#### **C.1.** Add 1 Placeholder to All Pages (similar to 8187a20)

> [!CAUTION]
> - Placeholder available on all pages.
> - Same content appears on News List **and** News Article pages.
> - Admin requires dev to move content (via template or CSS).

| Home | News |
| - | - |
| <img width="1194" alt="C1 An Extra Placholder - Home" src="https://github.com/user-attachments/assets/394e5d62-721e-48ac-828b-1e2880ac9ea7"> | <img width="1194" alt="C1 An Extra Placholder - News" src="https://github.com/user-attachments/assets/007072a0-05e1-4582-99b8-9044b12c0827"> |

### Solution B

> [!TIP]
> Ideal solution.

Full layout control for editor, likely via plugin that borrows from Blog plugin. Not achieved. Daunting. [Details (internal).](https://tacc-main.atlassian.net/browse/CMD-97?focusedCommentId=19702)

### Solution A

> [!CAUTION]
> - No pagination.

Include News via Blog CMS plugin for listing specific articles. Cancelled. Would not have pagination, but pagination is required. [Details (internal).](https://tacc-main.atlassian.net/browse/CMD-97?focusedCommentId=17359)
